apis/istio/v1alpha3/serviceentry_types.go

```diff
@@ -0,0 +1,390 @@
+// specified above. In other words, a call to http://foo.bar.com/baz would
+// be translated to http://uk.foo.bar.com/baz.
+//
+type ServiceEntrySpec struct {
+	// REQUIRED. The hosts associated with the ServiceEntry. Could be a DNS
+	// name with wildcard prefix (external services only). DNS names in hosts
+	// will be ignored if the application accesses the service over non-HTTP
+	// protocols such as mongo/opaque TCP/even HTTPS. In such scenarios, the
+	// IP addresses specified in the Addresses field or the port will be used
+	// to uniquely identify the destination.
+	Hosts []string `protobuf:"bytes,1,rep,name=hosts" json:"hosts,omitempty"`
+	// The virtual IP addresses associated with the service. Could be CIDR
+	// prefix.  For HTTP services, the addresses field will be ignored and
+	// the destination will be identified based on the HTTP Host/Authority
+	// header. For non-HTTP protocols such as mongo/opaque TCP/even HTTPS,
+	// the hosts will be ignored. If one or more IP addresses are specified,
+	// the incoming traffic will be identified as belonging to this service
+	// if the destination IP matches the IP/CIDRs specified in the addresses
+	// field. If the Addresses field is empty, traffic will be identified
+	// solely based on the destination port. In such scenarios, the port on
+	// which the service is being accessed must not be shared by any other
+	// service in the mesh. In other words, the sidecar will behave as a
+	// simple TCP proxy, forwarding incoming traffic on a specified port to
+	// the specified destination endpoint IP/host. Unix domain socket
+	// addresses are not supported in this field.
+	Addresses []string `protobuf:"bytes,2,rep,name=addresses" json:"addresses,omitempty"`
+	// REQUIRED. The ports associated with the external service. If the
+	// Endpoints are unix domain socket addresses, there must be exactly one
+	// port.
+	Ports []*Port `protobuf:"bytes,3,rep,name=ports" json:"ports,omitempty"`
+	// Specify whether the service should be considered external to the mesh
+	// or part of the mesh.
+	Location ServiceEntry_Location `protobuf:"varint,4,opt,name=location,proto3,enum=istio.networking.v1alpha3.ServiceEntry_Location" json:"location,omitempty"`
+	// REQUIRED: Service discovery mode for the hosts. Care must be taken
+	// when setting the resolution mode to NONE for a TCP port without
+	// accompanying IP addresses. In such cases, traffic to any IP on
+	// said port will be allowed (i.e. 0.0.0.0:<port>).
+	Resolution ServiceEntry_Resolution `protobuf:"varint,5,opt,name=resolution,proto3,enum=istio.networking.v1alpha3.ServiceEntry_Resolution" json:"resolution,omitempty"`
+	// One or more endpoints associated with the service.
+	Endpoints []*ServiceEntry_Endpoint `protobuf:"bytes,6,rep,name=endpoints" json:"endpoints,omitempty"`
+}
+
+// Endpoint defines a network address (IP or hostname) associated with
+// the mesh service.
+type ServiceEntry_Endpoint struct {
```

Comment:
The series of comments starting here regarding field naming seems to indicate you don't know where these fields are coming from:  protobuf itself.  All these field are named by protobuf.  
To override this and every comment after this will make it impossible to maintain.
Please look at the original [service_entry.pb.go](https://github.com/istio/api/blob/1.0.5/networking/v1alpha3/service_entry.pb.go).

Reply:
Comes from [serviceentry.pb.go#L454](https://github.com/istio/api/blob/1.0.5/networking/v1alpha3/service_entry.pb.go#L454)

Reply:
Thanks for explaining your process.  Please add comments into the code to make this clear.